imap_processing/codice/codice_l1a.py

```diff
@@ -106,24 +105,17 @@ def decompress_data(self, science_values: list[str]) -> None:
             compression_algorithm = constants.HI_COMPRESSION_ID_LOOKUP[self.view_id]
 
         self.raw_data = []
-        for i, packet_data in enumerate(science_values):
-            values = ast.literal_eval(
-                str(packet_data)
-            )  # convert from numpy array to byte object
-            # Some packets are not able to be decompressed because of packet
-            # continuation. For those that can't be processed, just use empty
-            # list for now
-            # TODO: Implement support for packet continuation (see issue #1155)
-            # TODO: Make sure all possible decompression errors are caught
-            try:
-                decompressed_values = decompress(values, compression_algorithm)
-            except lzma.LZMAError:
-                logger.warning(
-                    f'Cannot decompress {self.config["dataset_name"]} packet '
-                    f'{i+1} with compression algorithm {compression_algorithm}'
-                )
-                decompressed_values = []
+        for packet_data, byte_count in zip(
+            science_values, self.dataset.byte_count.data
```

Comment:
I was looking at what this `byte_count` data does/mean and I saw this description from the XTCE
> NUMBER OF BYTES IN THE DATA ARRAY.  IF COMPRESSED, THIS VALUE REPRESENTS THE LENGTH OF THE COMPRESSED DATA.

Looks like this `byte_count` is used by many CoDICE Hi and Lo packets. It says 'if compressed'. Then I looked at Lo and Hi's compression id lookup dictionary. It seems like all id maps to some compression form. Will there be a case when the data will not be compressed?

Also, I just noticed that in this [line](https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/12082088b587b21a32abb3592e8cd880f307dfb3/imap_processing/codice/codice_l1a.py#L416), you were using parameters from first packet for remaining all science packets. I think that may be remains from your previous implementation. Will that need to change now since `packet_data` from `science_values` may be using different view_id, plan_id, and so on.

Sorry for long comments. These comments are not for this PR but something I wanted to point out in case we missed it before.

Reply:
> Will there be a case when the data will not be compressed?

I haven't come across any data product yet that is expected to have no compression (algorithm=0) (see screenshot from the SCI_LUT spreadsheet). I think this is one of those knobs that CoDICE wants to be able to turn to theoretically, but in practice I am not sure it would ever happen. I will ask Joey about this though.[
<img width="508" alt="Screenshot 2024-11-21 at 11 01 29" src="https://github.com/user-attachments/assets/8ea4f530-db84-4ae0-8b64-8ddc8474673d">
](url)

> Also, I just noticed that in this [line](https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/12082088b587b21a32abb3592e8cd880f307dfb3/imap_processing/codice/codice_l1a.py#L416), you were using parameters from first packet for remaining all science packets. I think that may be remains from your previous implementation. Will that need to change now since packet_data from science_values may be using different view_id, plan_id, and so on.

I am a little confused here. Were you referring to this part of the code? https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/12082088b587b21a32abb3592e8cd880f307dfb3/imap_processing/codice/codice_l1a.py#L545, or event data processing?

If it is the former, I think it is safe to use the first packet because we expect each packet for a particular APID to have the same view_id, plan_step, etc., but I will also double check that with Joey.